mesh/mesh_test.go

```diff
@@ -363,15 +366,46 @@ func TestMesh_AddBlockWithTxs_PushTransactions_UpdateUnappliedTxs(t *testing.T)
 		r.Equal(111, int(txns[i].TotalAmount))
 	}
 
-	l, err := msh.GetLayer(1)
-	r.NoError(err)
-	msh.PushTransactions(l)
+	msh.pushLayersToState(1, 2)
+	r.Equal(4, len(state.Txs))
+
 	r.ElementsMatch(GetTransactionIds(tx5), GetTransactionIds(blockBuilder.txs...))
 
 	txns = getTxns(r, msh.MeshDB, origin)
 	r.Empty(txns)
 }
 
+func TestMesh_AddBlockWithTxs_PushTransactions_getInvalidBlocksByHare(t *testing.T) {
+	r := require.New(t)
+
+	msh := getMesh("mesh")
+
+	state := &MockMapState{}
+	msh.TxProcessor = state
+	blockBuilder := &MockBlockBuilder{}
+	msh.SetBlockBuilder(blockBuilder)
+
+	layerID := types.LayerID(1)
+	signer, _ := newSignerAndAddress(r, "origin")
+	tx1 := addTxToMesh(r, msh, signer, 2468)
+	tx2 := addTxToMesh(r, msh, signer, 2469)
+	tx3 := addTxToMesh(r, msh, signer, 2470)
+	tx4 := addTxToMesh(r, msh, signer, 2471)
+	tx5 := addTxToMesh(r, msh, signer, 2472)
+	var blocks []*types.Block
+	blocks = append(blocks, addBlockWithTxs(r, msh, layerID, true, tx1, tx2))
+	blocks = append(blocks, addBlockWithTxs(r, msh, layerID, true, tx2, tx3, tx4))
+	blocks = append(blocks, addBlockWithTxs(r, msh, layerID, true, tx4, tx5))
+	blocks = append(blocks, addBlockWithTxs(r, msh, layerID, true, tx5))
+	hareBlocks := blocks[:3]
+
+	invalid := msh.getInvalidBlocksByHare(types.NewExistingLayer(layerID, hareBlocks))
+	r.ElementsMatch(blocks[3:], invalid)
+
+	msh.reInsertTxsToPool(blocks[3:], invalid, layerID)
+
+}
```

Comment:
I think assertions are missing at this point? Otherwise we're only checking that this doesn't panic...

Reply:
good catch, I forgot to move this here
`r.ElementsMatch(GetTransactionIds(tx5), GetTransactionIds(invalidBlocks...))`
this is the part where this is supposed to be tested now, sinec validity of txs happens before the transaction flow